src/Faithlife.OAuth/OAuthUtility-HmacSha256.cs

```diff
@@ -0,0 +1,244 @@
+		{
+			List<KeyValuePair<string, string>> additionalParameters = new List<KeyValuePair<string, string>> { NewKeyValuePair(OAuthConstants.Token, accessToken) };
+			if (formParameters != null)
+				additionalParameters.AddRange(formParameters);
+			return CreateHmacSha256AuthorizationHeaderValue(uri, httpMethod, consumerSecret, accessSecret, GetHmacSha256Parameters(consumerToken, nonceCreator ?? s_nonceCreator, systemTime ?? s_systemTime, additionalParameters));
+		}
+
+		internal static string CreateHmacSha256Signature(string signatureBase, string consumerSecret, string? tokenSecret)
+		{
+			using var hmacsha256 = new HMACSHA256(Encoding.UTF8.GetBytes(CreatePlainTextSignature(consumerSecret, tokenSecret)));
+			return Convert.ToBase64String(hmacsha256.ComputeHash(Encoding.UTF8.GetBytes(signatureBase)));
+		}
+
+		private static string CreateHmacSha256AuthorizationHeaderValue(Uri uri, string httpMethod, string consumerSecret, string? tokenSecret, ICollection<KeyValuePair<string, string>> parameters)
+		{
+			string signatureBase = CreateSignatureBase(uri, httpMethod, parameters, out var newUri);
+
+			parameters.Add(new KeyValuePair<string, string>(OAuthConstants.Signature, CreateHmacSha256Signature(signatureBase, consumerSecret, tokenSecret)));
+
+			return OAuthConstants.HeaderPrefix + " " + parameters
+				.Where(parameter => parameter.Key.StartsWithOrdinal(OAuthConstants.ParameterPrefix) && parameter.Value != null)
+				.Select(p => "{0}=\"{1}\"".FormatInvariant(p.Key, PercentEncode(p.Value)))
+				.Join(",");
+		}
+
+		private static string CreateHmacSha256Signature(Uri uri, string httpMethod, string consumerSecret, string? tokenSecret, ICollection<KeyValuePair<string, string>> parameters)
+		{
+			return CreateHmacSha256Signature(CreateSignatureBase(uri, httpMethod, parameters, out string _), consumerSecret, tokenSecret);
+		}
+
+		private static ICollection<KeyValuePair<string, string>> GetHmacSha256Parameters(string consumerToken, INonceCreator nonceCreator, ISystemTime systemTime, IEnumerable<KeyValuePair<string, string>>? additionalParameters = null)
+		{
+			return GetHmacSha256Parameters(consumerToken, nonceCreator.CreateNonce(), DateTimeUtility.ToUnixTimestamp(systemTime.GetUtcNow()).ToInvariantString(), additionalParameters);
+		}
+
+		private static ICollection<KeyValuePair<string, string>> GetHmacSha256Parameters(string consumerToken, string nonce, string timeStamp, IEnumerable<KeyValuePair<string, string>>? additionalParameters = null)
+		{
+			return new List<KeyValuePair<string, string>>
+			{
+				NewKeyValuePair(OAuthConstants.ConsumerKey, consumerToken),
+				NewKeyValuePair(OAuthConstants.Nonce, nonce),
+				NewKeyValuePair(OAuthConstants.SignatureMethod, OAuthSignatureMethods.HmacSha256),
+				NewKeyValuePair(OAuthConstants.TimeStamp, timeStamp),
+				NewKeyValuePair(OAuthConstants.Version, OAuthConstants.OAuthVersion),
+			}.Union(additionalParameters.EmptyIfNull().Where(kvp => kvp.Value != null)).ToList();
+		}
```

Comment:
Seems like an exact duplication of https://github.com/Faithlife/FaithlifeOAuth/pull/3/files#diff-ad41888dfe9c00c72d737c029571963764aa9efd71c54af4218ced38c5937039R234-R241 except for `OAuthSignatureMethods.HmacSha256`. Let's parameterise on `OAuthSignatureMethods` and avoid duplication.

Reply:
Additionally, single-`return`-statement methods should use expression bodies.